bin/main.rb

```diff
@@ -1,2 +1,76 @@
 #!/usr/bin/env ruby
-puts 'Hello world!'
+
+puts 'WELCOME TO TIC-TAC-TOE GAME !!!'
+
+# create an empty game board
+@board = %w[1 2 3 4 5 6 7 8 9]
+
+# create game tokens as an array of 2 items
+@game_tokens = %w[X O]
+
+# create an empty hash to hold player and token info for each player
+# that is, player_one: 'X', player_two: 'O' or vice versa
+@assigned_tokens = {}
+
+# register two players
+puts 'Player 1, input name: '
+@player_one = gets.chomp
+
+puts 'Player 2, input name: '
+@player_two = gets.chomp
+
+# allow first player choose a game token X or O
+puts "#{@player_one}, Choose a token: 'X' or 'O'"
+selected_token = gets.chomp
+while selected_token
+  selected_token = selected_token.upcase
+
+  # validate the token chosen
+  if !@game_tokens.to_s.include? selected_token
+    puts "Invalid token, please select 'X' or 'O'"
```

Comment:
- [ ] [REQUIRED] Please take another look at the implementation for checking for a valid token. It results in an endless loop when an invalid token is chosen as shown below.
<img width="597" alt="Screen Shot 2020-10-01 at 4 48 50 PM" src="https://user-images.githubusercontent.com/18187039/94840037-5870c600-0407-11eb-9176-c8087f74d6b4.png">